modules/table/src/main/java/org/apache/ignite/internal/table/distributed/schema/PartitionCommandsMarshallerImpl.java

```diff
@@ -31,28 +32,40 @@ public PartitionCommandsMarshallerImpl(MessageSerializationRegistry serializatio
         super(serializationRegistry, cache);
     }
 
+    @Override
+    public void patch(ByteBuffer raw, HybridTimestamp safeTs) {
+        ByteBuffer dup = raw.duplicate().order(ORDER);
+        dup.putLong(4, safeTs.longValue());
+    }
+
     @Override
     protected void beforeWriteMessage(Object o, ByteBuffer buffer) {
         int requiredCatalogVersion = o instanceof CatalogVersionAware
                 ? ((CatalogVersionAware) o).requiredCatalogVersion()
                 : NO_VERSION_REQUIRED;
 
         stream.setBuffer(buffer);
-        stream.writeInt(requiredCatalogVersion);
+        stream.writeFixedInt(requiredCatalogVersion);
+        stream.writeFixedLong(0);
```

Comment:
Please add a comment saying that it allocates space for safe time

Reply:
🆗